pkg/collector/corechecks/cluster/kubernetes_eventbundle.go

```diff
@@ -56,17 +57,26 @@ func (k *kubernetesEventBundle) addEvent(event *v1.Event) error {
 
 	k.countByAction[fmt.Sprintf("**%s**: %s\n", event.Reason, event.Message)] += int(event.Count)
 	k.readableKey = fmt.Sprintf("%s %s", event.InvolvedObject.Name, event.InvolvedObject.Kind)
+
+	if event.InvolvedObject.Kind == "Node" || event.InvolvedObject.Kind == "Pod" {
+		k.hostname = event.Source.Host
+	}
+
 	return nil
 }
 
 func (k *kubernetesEventBundle) formatEvents(hostname string, modified bool) (metrics.Event, error) {
 	if len(k.events) == 0 {
 		return metrics.Event{}, errors.New("no event to export")
 	}
+	// If hostname was not defined (event not specific to a host), use the local hostname
```

Comment:
empty string "" is a valid hostname value for events sent from checks. The aggregator will automatically add the agent hostname later.
Can we skip this step? BTW we might later tell the aggregator to leave an empty hostname for these when it'll be running as a cluster check. So I'd go with keeping the empty string

Reply:
Oh, didn't know that the hostname would be added automatically. Sure, this makes sense

Reply:
👍